### PR TITLE
Show projected APY rate

### PR DESCRIPTION
**Pull request by RodionChachura**
image

### CODE REVIEW

File: app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  if (yieldReserveChange.lt(dynRateMaxChange)) {
+    yieldReserveChange = dynRateMaxChange;
+  }
+
+  let projectedEarnApy = big(0);
+  if (isCurrentYieldReserveHigher) {
+    projectedEarnApy = currentRate.plus(yieldReserveChange);
+  } else if (currentRate.lt(yieldReserveChange)) {
```
Comment (cosullivan) [inline app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts:87]: @RodionChachura this is the issue here, this should be currentRate.gt(...) not .lt()

Comment (cosullivan) [reply app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts:87]: @RodionChachura also see if you can split this out into smaller functions to improve readability, ie,

const { change, isHigher } = computeYieldReserveChange(...)

const projected = computeProjectedApr(...)

Comment (RodionChachura) [reply app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts]: 🤦‍♂️✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/309/commits/678de5c365b9ec42aff264447845a4a1320a4cbe
image

Comment (simke9445) [reply app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts]: As a user I'm not really sure through the UI what the "Projected 19.97%" means here. 
I think we should add a tooltip explaining what the projection means.

Comment (simke9445) [reply app/src/@anchor-protocol/app-provider/queries/earn/useProjectedEarnApy.ts]: As a user I'm not really sure through the UI what the "Projected 19.97%" means here. 
I think we should add a tooltip explaining what the projection means.

**Comment by cosullivan** [comment]
The styling change looks good, I'm just reviewing now.

File: app/src/queries/useEstimatedTimestamp.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  blockHeight: number,
+): number | undefined => {
+  const {
+    constants: { blocksPerYear },
+  } = useAnchorWebapp();
+
+  const { data: lastSyncedBlock } = useLastSyncedBlock();
+  if (!lastSyncedBlock) return;
+
+  const daysInYear = getDaysInYear(new Date());
```
Comment (RodionChachura) [inline app/src/queries/useEstimatedTimestamp.ts:17]: It won't be super corrected for a block that is not in the same year, but it's not significant in the current use case.

File: app/src/queries/useLastSyncedBlock.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  return {
+    height: Number(height),
+    timestamp: new Date(time).getTime(),
+  };
+};
+
+const lastSyncedBlockQueryFn = createQueryFn(lastSyncedBlockQuery);
+
+export const useLastSyncedBlock = (): UseQueryResult<Block> => {
```
Comment (RodionChachura) [inline app/src/queries/useLastSyncedBlock.ts:43]: I left `useLastSyncedHeight` unchanged to not blew up the current PR.

File: app/src/queries/useEstimatedTimestamp.ts (hunk 2 above)
Comment (cosullivan) [reply app/src/queries/useEstimatedTimestamp.ts:17]: I'm not a fan omitting the braces, at some point I will add that to the linting rules so we may aswell add it now, so

`if (!lastSyncedBlock) return;` should be

```
if (!lastSyncedBlock) {
  return;
}
```

Also I avoid using the negating operator where possible as its too easy to miss when reading code, so this is way more explicit

```
if (lastSyncedBlock === undefined) {
  return;
}
```

**Comment by cosullivan** [comment]
@RodionChachura can we change the tooltip wording to be 

"Projected annualized deposit rate from April 4"

File: app/src/queries/useEstimatedTimestamp.ts (hunk 2 above)
Comment (RodionChachura) [reply app/src/queries/useEstimatedTimestamp.ts:19]: ✅ https://github.com/Anchor-Protocol/anchor-web-app/pull/309/commits/a39b86bccd46575eee5ea009a7eaae3f392f7fba

**Comment by RodionChachura** [comment]
> Projected annualized deposit rate from

✅
image

**Pull request merged**